packages/rocketchat-lib/server/methods/leaveRoom.js

```diff
@@ -27,7 +26,17 @@ Meteor.methods({
 				throw new Meteor.Error('error-you-are-last-owner', 'You are the last owner. Please set new owner before leaving the room.', { method: 'leaveRoom' });
 			}
 		}
-
+		if (room.automatic) {
+
+			RocketChat.leave_automatic_channel(room.name);
+			// delete room if its the last user
```

Comment:
There is a method but that doesn't allow the deletion of a channel when a normal user is logged in. It shows an error.